native/exe/ServiceControlLaunch/ServiceControlLaunch.cpp

```diff
@@ -105,9 +109,11 @@ void popupExceptionError(JNIEnv* env, jthrowable thrower, char* errTitle)
 	jstring throwStr = (jstring)env->CallObjectMethod(thrower, toStr);
 	const char* cThrowStr = env->GetStringUTFChars(throwStr, 0);
 	char *errStr = (char*) malloc(env->GetStringLength(throwStr) + 64);
-	sprintf(errStr, "An exception occured in Java:\n%s", cThrowStr);
+	size_t errStrlen = strlen(cThrowStr) + 64;  // Need to know this size for the sprintf_s

```

Comment:
This code I wrote is so ugly...but I wrote it many years ago. :)  A cleaner way here is to establish the length before the call to malloc and then pass that length in there. Then you can reuse that length in snprintf to ensure buffer safety.

Reply:
Done